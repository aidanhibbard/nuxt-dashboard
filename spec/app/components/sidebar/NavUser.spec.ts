import { describe, it, expect, vi, beforeEach } from 'vitest'
import { mount } from '@vue/test-utils'
import NavUser from '@/components/navigation/sidebar/NavUser.vue'

const stubs = {
  Avatar: { template: '<div class="avatar"><slot /></div>' },
  AvatarFallback: { template: '<div class="fallback"><slot /></div>' },
  AvatarImage: { template: '<img />' },
  DropdownMenu: { template: '<div><slot /></div>' },
  DropdownMenuContent: { template: '<div><slot /></div>' },
  DropdownMenuGroup: { template: '<div><slot /></div>' },
  DropdownMenuItem: { template: '<div><slot /></div>' },
  DropdownMenuLabel: { template: '<div><slot /></div>' },
  DropdownMenuSeparator: { template: '<hr />' },
  DropdownMenuTrigger: { template: '<div><slot /></div>' },
  SidebarMenu: { template: '<ul><slot /></ul>' },
  SidebarMenuButton: { template: '<button><slot /></button>' },
  SidebarMenuItem: { template: '<li><slot /></li>' },
}

// useSidebar is provided by setup file mock

describe('components/navigation/sidebar/NavUser.vue', () => {
  const user = { name: 'Alice', email: 'alice@example.com', avatar: '/alice.png' }

  it('renders user name and email', () => {
    const wrapper = mount(NavUser, { global: { stubs }, props: { user } })
    expect(wrapper.text()).toContain('Alice')
    expect(wrapper.text()).toContain('alice@example.com')
  })

  it('matches snapshot (desktop)', () => {
    ;(globalThis as any).__sidebarIsMobile = false
    const wrapper = mount(NavUser, { global: { stubs }, props: { user } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('changes dropdown side on mobile', () => {
    ;(globalThis as any).__sidebarIsMobile = true
    const wrapper = mount(NavUser, { global: { stubs }, props: { user } })
    expect(wrapper.find('[data-side="bottom"]').exists()).toBe(true)
  })
})


