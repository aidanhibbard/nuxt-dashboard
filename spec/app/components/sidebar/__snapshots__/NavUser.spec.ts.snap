// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/navigation/sidebar/NavUser.vue > matches snapshot (desktop) 1`] = `
"<ul>
  <li>
    <div>
      <div as-child=""><button size="lg" class="data-[state=open]:bg-sidebar-accent data-[state=open]:text-sidebar-accent-foreground">
          <div class="avatar h-8 w-8 rounded-lg"><img src="/alice.png" alt="Alice">
            <div class="fallback rounded-lg"> CN </div>
          </div>
          <div class="grid flex-1 text-left text-sm leading-tight"><span class="truncate font-medium">Alice</span><span class="truncate text-xs">alice@example.com</span></div><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide ml-auto size-4 lucide-chevrons-up-down-icon lucide-chevrons-up-down ml-auto size-4">
            <path d="m7 15 5 5 5-5"></path>
            <path d="m7 9 5-5 5 5"></path>
          </svg>
        </button></div>
      <div data-side="right" data-align="end" class="w-[--reka-dropdown-menu-trigger-width] min-w-56 rounded-lg" side-offset="4">
        <div class="p-0 font-normal">
          <div class="flex items-center gap-2 px-1 py-1.5 text-left text-sm">
            <div class="avatar h-8 w-8 rounded-lg"><img src="/alice.png" alt="Alice">
              <div class="fallback rounded-lg"> CN </div>
            </div>
            <div class="grid flex-1 text-left text-sm leading-tight"><span class="truncate font-semibold">Alice</span><span class="truncate text-xs">alice@example.com</span></div>
          </div>
        </div>
        <hr>
        <div>
          <div class="dropdown-menu-item"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-sparkles-icon lucide-sparkles">
              <path d="M11.017 2.814a1 1 0 0 1 1.966 0l1.051 5.558a2 2 0 0 0 1.594 1.594l5.558 1.051a1 1 0 0 1 0 1.966l-5.558 1.051a2 2 0 0 0-1.594 1.594l-1.051 5.558a1 1 0 0 1-1.966 0l-1.051-5.558a2 2 0 0 0-1.594-1.594l-5.558-1.051a1 1 0 0 1 0-1.966l5.558-1.051a2 2 0 0 0 1.594-1.594z"></path>
              <path d="M20 2v4"></path>
              <path d="M22 4h-4"></path>
              <circle cx="4" cy="20" r="2"></circle>
            </svg> Upgrade to Pro </div>
        </div>
        <hr>
        <div>
          <div class="dropdown-menu-item"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-badge-check-icon lucide-badge-check">
              <path d="M3.85 8.62a4 4 0 0 1 4.78-4.77 4 4 0 0 1 6.74 0 4 4 0 0 1 4.78 4.78 4 4 0 0 1 0 6.74 4 4 0 0 1-4.77 4.78 4 4 0 0 1-6.75 0 4 4 0 0 1-4.78-4.77 4 4 0 0 1 0-6.76Z"></path>
              <path d="m9 12 2 2 4-4"></path>
            </svg> Account </div>
          <div class="dropdown-menu-item"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-credit-card-icon lucide-credit-card">
              <rect width="20" height="14" x="2" y="5" rx="2"></rect>
              <line x1="2" x2="22" y1="10" y2="10"></line>
            </svg> Billing </div>
          <div class="dropdown-menu-item"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-bell-icon lucide-bell">
              <path d="M10.268 21a2 2 0 0 0 3.464 0"></path>
              <path d="M3.262 15.326A1 1 0 0 0 4 17h16a1 1 0 0 0 .74-1.673C19.41 13.956 18 12.499 18 8A6 6 0 0 0 6 8c0 4.499-1.411 5.956-2.738 7.326"></path>
            </svg> Notifications </div>
        </div>
        <hr>
        <div class="dropdown-menu-item"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-log-out-icon lucide-log-out">
            <path d="m16 17 5-5-5-5"></path>
            <path d="M21 12H9"></path>
            <path d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"></path>
          </svg> Log out </div>
      </div>
    </div>
  </li>
</ul>"
`;
